Permissions corrupted in gen-sdk macosx headers

I think #24534 broke the MacOS SDK:
```
guest@gnu /tmp/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers$ ls -al
total 32
drwxr-xr-x 4 guest users 4096 Apr 22 18:58 ./
drwxrwxrwt 4 root  root  4096 Apr 22 18:58 ../
---x---r-- 1 guest users  127 Jan  1  1970 Entitlements.plist
---x---r-- 1 guest users 4307 Jan  1  1970 SDKSettings.json
---x---r-- 1 guest users 3597 Jan  1  1970 SDKSettings.plist
drwxr-xr-x 4 guest users 4096 Jan  1  1970 System/
drwxr-xr-x 7 guest users 4096 Jan  1  1970 usr/
```
Causing the guix build to fail with:
```
checking for x86_64-apple-darwin-gcc... env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/kvaciral/.guix-profile/bin/clang --target=x86_64-apple-darwin -mmacosx-version-min=10.15 -B/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=609 -isysroot/bitcoin/depends/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers -Xclang -internal-externc-isystem/gnu/store/v770rvqs8q21mbzwb3gkihr2glgn80am-clang-10.0.1/lib/clang/10.0.1/include -Xclang -internal-externc-isystem/bitcoin/depends/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers/usr/include
checking whether the C compiler works... no
configure: error: in `/bitcoin/depends/work/build/x86_64-apple-darwin/libevent/2.1.12-stable-6395cab7fb6':
configure: error: C compiler cannot create executables
See `config.log' for more details
make: *** [[funcs.mk:283](http://funcs.mk:283/): /bitcoin/depends/work/build/x86_64-apple-darwin/libevent/2.1.12-stable-6395cab7fb6/./.stamp_configured] Error 77
make: Leaving directory '/bitcoin/depends'
kvaciral@penumbra:~/projects/bitcoin$
```

Reported by @kvaciral.

--- a/contrib/macdeploy/README.md
+++ b/contrib/macdeploy/README.md
@@ -58,7 +58,7 @@ previous stage) as the first argument.
 ./contrib/macdeploy/gen-sdk '/path/to/Xcode.app'
 ```
 
-The `sha256sum` of the generated TAR.GZ archive should be `e7ca56bc8804d16624fad68be2e71647747d6629cacaaa3de5fbfa7f444e9eae`.
+The `sha256sum` of the generated TAR.GZ archive should be `df75d30ecafc429e905134333aeae56ac65fac67cb4182622398fd717df77619`.
 
 ## Deterministic macOS DMG Notes
 

--- a/contrib/macdeploy/gen-sdk
+++ b/contrib/macdeploy/gen-sdk
@@ -95,7 +95,7 @@ def run():
             tarinfo.uid, tarinfo.uname =  0, ''
             tarinfo.gid, tarinfo.gname = 0, ''
             # don't use isdir() as there are also executable files present
-            tarinfo.mode = 0o0755 if tarinfo.mode & 0o0100 else 0x0644
+            tarinfo.mode = 0o0755 if tarinfo.mode & 0o0100 else 0o0644
             return tarinfo
         with cd(dir_to_add):
             # recursion already adds entries in sorted order

